include/lug/System/Memory/Allocator/Linear.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <cstddef>
+#include <lug/System/Export.hpp>
+#include <lug/System/Memory/Area/IArea.hpp>
+
+namespace lug {
+namespace System {
+namespace Memory {
+namespace Allocator {
+
+class LUG_SYSTEM_API Linear {
+public:
+    struct Mark {
+        void* current{nullptr};
+        lug::System::Memory::Area::Page* currentPage{nullptr};
+    };
+
+public:
+    Linear(lug::System::Memory::Area::IArea* area);
+    Linear(const Linear&) = delete;
+    Linear(Linear&&) = default;
+
+    Linear& operator=(const Linear&) = delete;
+    Linear& operator=(Linear&&) = default;
+
+    ~Linear() = default;
+
+    void* allocate(size_t size, size_t alignment, size_t offset);
+    void free(void* ptr) const;
+    void reset();
+
+    // Dangerous operations
```

Comment:
Hum ... That seems obvious, if you reset the allocator from a previous state and free the value after this mark that won't works.